src/js/model/string/string.js

```diff
@@ -0,0 +1,43 @@
+var AMPParse = AMPParse || {};
+
+AMPParse.buildString = function (hexadecimal) {
+    nibblesUsed = 0; //int
+    nullTerminate = false; //bool
+    returnString = ""; //string
+    tempChars = "00";
+    
+    // Defensive programming
+    if (typeof hexadecimal === "undefined" || typeof hexadecimal !== "string") {
+	throw new ReferenceError("Provided parameter does not exist or is not a string");
+    }
+    tempChars = hexadecimal.substring(0, 2);
+    if (tempChars == "00") {
+	throw new RangeError ("The provided hex is too short to contain a string");
+    }
+
+    // Find length of string
+    while (!nullTerminate) {
+	tempChars = hexadecimal.substring(nibblesUsed, nibblesUsed+2);
+	if (tempChars != "00") {
+ 	    nibblesUsed++;
+	    nibblesUsed++;
+	    returnString += tempChars;
```

Comment:
You'll need to convert those bytes that you're pulling into actual characters (i.e. 0x41 = 'A')